### PR TITLE
Refactored the repositories API to use the new pattern

### PR DESCRIPTION
**Pull request by dhruvsinghal**
@pengwynn please take a look. (Note that I would be including the collaborators API in a different PR).

### CODE REVIEW

File: octokit/repositories.go [hunk 1]
```diff
@@ -1,57 +1,91 @@
...
 // Hyperlinks to the various repository locations on github.
-// RepositoryURL is a template for a particular repository for a particular owner.
-// ForksURL is a template for the forks of a user's repository.
-// UserRepositoriesURL is the address for all user repositories.
-// OrgRepositoriesUrl is the template for repositories within a particular organization.
 var (
-	RepositoryURL       = Hyperlink("repos/{owner}/{repo}")
-	ForksURL            = Hyperlink("repos/{owner}/{repo}/forks")
+	// RepositoryURL is a template for a particular repository for a
+	// particular owner.
```
Comment (pengwynn) [inline octokit/repositories.go]: I think @dannysperling had to break these out into multiple `var` statements to get them to play nice with GoDoc but I could be wrong.

File: octokit/repositories.go [hunk 2]
```diff
@@ -1,57 +1,91 @@
...
 	return
 }
 
 // All gets a list of all repositories associated with the url of the service
-func (r *RepositoriesService) All() (repos []Repository, result *Result) {
-	result = r.client.get(r.URL, &repos)
+func (r *RepositoriesService) All(uri *Hyperlink, params M) (repos []Repository,
+	result *Result) {
+	if uri == nil {
+		uri = &RepositoryURL
```
Comment (pengwynn) [inline octokit/repositories.go]: Shouldn't this default to a listing resource URL like `/repositories` (list all public repos) or `/user/repos` (list mine)?

File: octokit/repositories.go (hunk 1 above)
Comment (dannysperling) [inline octokit/repositories.go]: Pretty sure they get grouped together as a single element in the godoc, and the godoc is the comments outside the `var`.

Comment (dhruvsinghal) [inline octokit/repositories.go]: Fixed.

File: octokit/repositories.go (hunk 2 above)
Comment (dhruvsinghal) [inline octokit/repositories.go]: Fixed.

**Comment by dhruvsinghal** [comment]
@pengwynn Please take another look.

**Comment by dhruvsinghal** [comment]
@pengwynn Please take another look.

**Comment by dhruvsinghal** [comment]
@pengwynn

File: octokit/repositories.go [hunk 3]
```diff
@@ -1,57 +1,89 @@
...
 
 // All gets a list of all repositories associated with the url of the service
-func (r *RepositoriesService) All() (repos []Repository, result *Result) {
-	result = r.client.get(r.URL, &repos)
+func (r *RepositoriesService) All(uri *Hyperlink, params M) (repos []Repository,
+	result *Result) {
+	if uri == nil && len(params) == 0 {
+		uri = &UserRepositoriesURL
+	} else if uri == nil {
+		uri = &RepositoryURL
```
Comment (pengwynn) [inline octokit/repositories.go]: I wonder if this second fallback should be [this URI][uri] instead. 

[uri]: https://developer.github.com/v3/repos/#list-all-public-repositories

File: octokit/repositories.go [hunk 4]
```diff
@@ -1,57 +1,90 @@
...
-	URL    *url.URL
 }
 
 // One gets a specific repository based on the url of the service
-func (r *RepositoriesService) One() (repo *Repository, result *Result) {
-	result = r.client.get(r.URL, &repo)
+func (r *RepositoriesService) One(uri *Hyperlink, params M) (repo *Repository,
+	result *Result) {
+	if uri == nil {
+		uri = &AllRepositoriesURL
```
Comment (pengwynn) [inline octokit/repositories.go]: Since this is `One()`, should this be `RepositoryURL`?

File: octokit/repositories_test.go [hunk 5]
```diff
@@ -97,9 +98,11 @@ func TestRepositoresService_Create(t *testing.T) {
 	assert.Equal(t, "This is your first repo", repo.Description)
 	assert.False(t, repo.Private)
 	assert.True(t, repo.Fork)
-	assert.Equal(t, "https://api.github.com/repos/octocat/Hello-World", repo.URL)
+	assert.Equal(t, "https://api.github.com/repos/octocat/Hello-World",
```
Comment (pengwynn) [inline octokit/repositories_test.go]: Based on testify's recent changes, I think we need `EqualValues` when comparing a string to a `Hyperlink`, right?

**Comment by pengwynn** [comment]
A couple of small comments on this one @dhruvsinghal.

**Comment by dhruvsinghal** [comment]
@pengwynn Made the changes. Please take a look.

**Pull request merged**